gilmour.go

```diff
@@ -523,16 +536,14 @@ func isExpired(timeout time.Duration, startTime time.Time) bool {
 	return time.Now().After(startTime.Add(timeout))
 }
 
-func try(r RetryConf, fn func() error) (err error) {
+func try(g *Gilmour, fn func() error) (err error) {
 	startTime := time.Now()
 	for {
 		err = fn()
-		if isExpired(r.Timeout, startTime) || !isNetworkError(err) || err == nil {
+		if isExpired(g.getRetryTimeout(), startTime) || !isNetworkError(err) || err == nil {
```

Comment:
@meson10 Operation needs to performed atleast once, In any case I will be testing for timeout after the actual operation. What difference does that make if I do that out of the function?